File: src/models/ticket.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct JiraTicket {
    pub key: String,
    pub fields: TicketFields,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct TicketFields {
    pub summary: String,
    #[serde(default)]
    pub description: Option<String>,
    pub status: Status,
    #[serde(default)]
    pub assignee: Option<User>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct Status {
    pub name: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct User {
    #[serde(rename = "displayName")]
    pub display_name: String,
}

File: internal/models/ticket.go
package models

type JiraTicket struct {
	Key    string       `json:"key"`
	Fields TicketFields `json:"fields"`
}

type TicketFields struct {
	Summary     string       `json:"summary"`
	Description string       `json:"description"`
	Status      TicketStatus `json:"status"`
	Assignee    *TicketUser  `json:"assignee"`
}

type TicketStatus struct {
	Name string `json:"name"`
}

type TicketUser struct {
	DisplayName string `json:"displayName"`
}
